std_execution.bs

```diff
@@ -1209,13 +1227,34 @@ Returns a sender which, when started, will start the provided sender on an execu
 ### `execution::into_variant` ### {#design-sender-adaptor-into_variant}
 
 <pre highlight=c++>
-execution::sender into_variant(
+execution::sender auto into_variant(
     execution::sender auto snd
 );
 </pre>
 
 Returns a sender which sends a variant of tuples of all the possible sets of types sent by the input sender. Senders can send multiple sets of values depending on runtime conditions; this is a helper function that turns them into a single variant value.
 
+### `execution::done_as_optional` ### {#design-sender-adaptor-done_as_optional}
+
+<pre highlight=c++>
+execution::sender auto done_as_optional(
+    execution::sender auto snd
+);
+</pre>
+
+Returns a sender that maps the value channel from a `T` to an `optional<T>`, and maps the done channel to a value of an empty `optional<T>`.
```

Comment:
Should we mention that this requires a single-type sender here?

Reply:
Can this also elaborate on what it does when `T` is a reference-type?
Does it decay-copy `T` into an `optional<remove_cvref_t<T>>` or does it wrap references into `optional<reference_wrapper<T>>`?

Reply:
I would say `reference_wrapper`, except that doesn't work with rvalue references. Probably decay-copy for the time being.